brainstorming.md

```diff
@@ -0,0 +1,557 @@
+| delete(uri)         | Delete the SOMACollection specified with the URI.                              |
+| exists(uri) -> bool | Return true if object exists and is a SOMACollection.                          |
+| get metadata        | Access the metadata as a mutable [`SOMAMetadataMapping`](#SOMAMetadataMapping) |
+| get type            | Returns the constant "SOMACollection"                                          |
+
+In addition, SOMACollection supports operations to manage the contents of the collection:
+
+| Operation                        | Description                                   |
+| -------------------------------- | --------------------------------------------- |
+| get(string key)                  | Get the object associated with the key        |
+| has(string key)                  | Test for the existence of key in collection.  |
+| set(string key, ValueType value) | Set the key/value in the collection.          |
+| del(string key)                  | Remove the key/value from the collection.     |
+| iterator                         | Iterate over the collection.                  |
+| get length                       | Get the number of elements in the collection. |
+
+> ⚠️ Issue - does the delete() operation recursively delete, or simply remove the container? For now, define it as removing from container, but not deleting the underlying object.
+
+## SOMADataFrame
+
+> ⚠️ **To be further specified** -- all methods need specification.
+
+Summary of operations:
+
+| Operation                               | Description                                                                             |
+| --------------------------------------- | --------------------------------------------------------------------------------------- |
+| create(uri, ...)                        | Create a SOMADataFrame.                                                                 |
+| delete(uri)                             | Delete the SOMADataFrame specified with the URI.                                        |
+| exists(uri) -> bool                     | Return true if object exists and is a SOMADataFrame.                                    |
+| get metadata                            | Access the metadata as a mutable [`SOMAMetadataMapping`](#SOMAMetadataMapping)          |
+| get type                                | Returns the constant "SOMADataFrame"                                                    |
+| get shape -> (int, ...)                 | Return length of each dimension, always a list of length `ndims`                        |
+| get ndims -> int                        | Return number of index columns                                                          |
+| get schema -> Arrow.Schema              | Return data schema, in the form of an Arrow Schema                                      |
+| get is user-index -> bool               | Return true if user-indexed, false if row-indexed.                                      |
```

Comment:
`get_is_user_indexed`

Reply:
I cleaned these up so that they are consistent in the doc.  TY!